File: src/main/java/nl/guno/intellij/savechangelisttopatches/settings/SettingsManager.java
package nl.guno.intellij.savechangelisttopatches.settings;

import javax.swing.*;

import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import com.intellij.openapi.options.ConfigurationException;
import com.intellij.openapi.options.SearchableConfigurable;
import com.intellij.openapi.project.Project;
import nl.guno.intellij.savechangelisttopatches.MessageResources;

public class SettingsManager implements SearchableConfigurable {
    private final Project myProject;
    private SettingsPanel mySettingsPane;

    public SettingsManager(Project project) {
        myProject = project;
    }

    @Nls
    @Override
    public String getDisplayName() {
        return MessageResources.message("settingsManager.displayName");
    }

    @Nullable
    @Override
    public String getHelpTopic() {
        return null;
    }

    @NotNull
    @Override
    public String getId() {
        return "saveChangeListPatches";
    }

    @Nullable
    @Override
    public Runnable enableSearch(String option) {
        return null;
    }

    @Nullable
    @Override
    public JComponent createComponent() {
        if (mySettingsPane == null) {
            mySettingsPane = new SettingsPanel(myProject);
        }
        return mySettingsPane.getPanel();
    }

    @Override
    public boolean isModified() {
        return mySettingsPane != null && mySettingsPane.isModified();
    }

    @Override
    public void apply() throws ConfigurationException {
        if (mySettingsPane != null) {
            mySettingsPane.apply();
        }
    }

    @Override
    public void reset() {
        if (mySettingsPane != null) {
            mySettingsPane.reset();
        }
    }

    @Override
    public void disposeUIResources() {
    }
}

File: src/main/kotlin/nl/guno/intellij/savechangelisttopatches/settings/SettingsManager.kt
package nl.guno.intellij.savechangelisttopatches.settings

import javax.swing.*

import org.jetbrains.annotations.Nls

import com.intellij.openapi.options.ConfigurationException
import com.intellij.openapi.options.SearchableConfigurable
import com.intellij.openapi.project.Project
import nl.guno.intellij.savechangelisttopatches.MessageResources

class SettingsManager(private val myProject: Project) : SearchableConfigurable {
    private var mySettingsPane: SettingsPanel? = null

    @Nls
    override fun getDisplayName(): String {
        return MessageResources.message("settingsManager.displayName")
    }

    override fun getHelpTopic(): String? {
        return null
    }

    override fun getId(): String {
        return "saveChangeListPatches"
    }

    override fun enableSearch(option: String): Runnable? {
        return null
    }

    override fun createComponent(): JComponent? {
        if (mySettingsPane == null) {
            mySettingsPane = SettingsPanel(myProject)
        }
        return mySettingsPane!!.panel
    }

    override fun isModified(): Boolean {
        return mySettingsPane != null && mySettingsPane!!.isModified
    }

    @Throws(ConfigurationException::class)
    override fun apply() {
        if (mySettingsPane != null) {
            mySettingsPane!!.apply()
        }
    }

    override fun reset() {
        if (mySettingsPane != null) {
            mySettingsPane!!.reset()
        }
    }

    override fun disposeUIResources() {}
}
